event_groups.c

```diff
@@ -625,43 +625,49 @@ EventBits_t xEventGroupSetBits( EventGroupHandle_t xEventGroup,
 
 void vEventGroupDelete( EventGroupHandle_t xEventGroup )
 {
-    EventGroup_t * pxEventBits = xEventGroup;
-    const List_t * pxTasksWaitingForBits = &( pxEventBits->xTasksWaitingForBits );
+    EventGroup_t * pxEventBits;
+    const List_t * pxTasksWaitingForBits;
 
-    vTaskSuspendAll();
+    if ( NULL != xEventGroup )

```

Comment:
It might be best to add a configASSERT ( xEventGroup ) so that the error condition can be reported during development instead of silently failing.

Reply:
I considered of that but I think not using assert is more common cases of a funciton that free a memory with no side effects. I don't think it's a good idea to break the code running by this.